### DISCUSSION TITLE
How do you get the Cost price on the energy dash board so it can be used in ESPhome?

### DISCUSSION THREAD

**Discussion by anothermort** [BottlecapDave/HomeAssistant-OctopusEnergy#377] [Q&A] [answered]
Hi
I currently have a ESPhome Octo Mini display using an Esp32 and a Waveshare 2.9" E-Paper Display powered by BottlecapDave's HA OctopusEnergy integration. It is showing up the HA data very nicely in kWh for both gas and electricity.

Very new to HA and ESPhome. I would like to display the current total cost of electricity and gas used so far today.

But am unsure how to get the data from the energy dashboard on HA into ESPhome

Is it addressed as an entity? Or something completely different?

Not sure what to look for.

Any help appreciated. Thank you

Adam

**Comment by BottlecapDave** [comment, marked as answer]
Hello.

Depending on how you've set up the energy dashboard, it would be recommended to use either the [current accumulative cost](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/sensors/electricity.md#current-accumulative-cost) (if you have an Octopus Mini) or the [previous accumulative cost](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/sensors/electricity.md#previous-accumulative-cost) (for the previous day) to get the total cost.

I don't believe the raw value calculated in the energy dashboard is exposed as an entity in any way.

**Comment by anothermort** [comment]
Awesome. Thank you.
